src/test/java/guitests/FilterTextFieldTest.java

```diff
@@ -121,6 +62,96 @@ public void cancelTest() {
         awaitCondition(() -> toggle.get() % 2 == 1);
     }
 
+    @Test
+    public void filterTextFieldColor_validFilter_validFilterStyleApplied(){
+        FilterTextField field = getFirstPanelField();
+
+        clearField();
+        type("is:open");
+        assertTrue(field.getStyle().contains(VALID_FILTER_STYLE));
+
+        clearField();
+        type("-is:open");
+        assertTrue(field.getStyle().contains(VALID_FILTER_STYLE));
+
+        clearField();
+        type("hello");
+        assertTrue(field.getStyle().contains(VALID_FILTER_STYLE));
+
+        clearField();
+        type("is:open & has:assignee");
+        assertTrue(field.getStyle().contains(VALID_FILTER_STYLE));
+
+        clearField();
+        type("(is:closed)");
+        assertTrue(field.getStyle().contains(VALID_FILTER_STYLE));
+    }
+
+    @Test
+    public void filterTextFieldColor_invalidFilter_invalidFilterStyleApplied(){
+        FilterTextField field = getFirstPanelField();
+
+        // Tests if parse errors applies invalid filter style to text field
+        clearField();
+        type("is:-open");
+        assertTrue(field.getStyle().contains(INVALID_FILTER_STYLE));
+
+        clearField();
+        type("--");
+        assertTrue(field.getStyle().contains(INVALID_FILTER_STYLE));
+
+        clearField();
+        type("is:-open & has:assignee");
+        assertTrue(field.getStyle().contains(INVALID_FILTER_STYLE));
+
+        clearField();
+        type("(is:closed");
+        assertTrue(field.getStyle().contains(INVALID_FILTER_STYLE));
+
+        // Tests if semantic errors applies invalid filter style to text field
+        // These invalid filters will cause the invalid filter style be applied after ENTER is pressed
+        clearField();
+        type("is:invalid").push(KeyCode.ENTER);
+        assertTrue(field.getStyle().contains(INVALID_FILTER_STYLE));
+
+        clearField();
+        type("is:invalid & hello").push(KeyCode.ENTER);
+        assertTrue(field.getStyle().contains(INVALID_FILTER_STYLE));
+    }
+
+    private void testSpaces(FilterTextField field) {
```

Comment:
Were there any changes to this and the one below? Can't tell since the whole thing is in the diff.